internal/backend/projects/lint.go

```diff
@@ -0,0 +1,108 @@
+package projects
+
+import (
+	"fmt"
+
+	"go.autokitteh.dev/autokitteh/internal/backend/projectsgrpcsvc"
+	"go.autokitteh.dev/autokitteh/internal/manifest"
+	"go.autokitteh.dev/autokitteh/sdk/sdktypes"
+)
+
+type Checker func(projectID sdktypes.ProjectID, manifest *manifest.Manifest, resources map[string][]byte) []*sdktypes.CheckViolation
```

Comment:
I think you can wrap the inputs in a type
it will be easier to extend later and it makes all the checker signatures simple, instead of stuff like

```go
func checkNoTriggers(_ sdktypes.ProjectID, m *manifest.Manifest, _ map[string][]byte) []*sdktypes.CheckViolation
```

or 
```go
func checkEmptyVars(_ sdktypes.ProjectID, m *manifest.Manifest, _ map[string][]byte) []*sdktypes.CheckViolation
```

you can do 
```go
type ProjectDescriptor struct {
 pid sdktypes.ProjectID
 resources map[string][]byte
 manifest manifest.Manifest
}

type Checker func(desc ProjectDescriptor) []*sdktypes.CheckViolation

func Validate(projectID sdktypes.ProjectID, manifest *manifest.Manifest, resources map[string][]byte) []*sdktypes.CheckViolation {
 	var vs []*sdktypes.CheckViolation
        desc := ProjectDescriptor{
              pid: projectID,
              manifest: manifest,
              resources: resources
        }
 	for _, checker := range lintCheckers {
 		vs = append(vs, checker(desc)...)
 	}

 	return vs
 }
```

Reply:
not sure I agree, but ok